### ISSUE TITLE
Color Space Explorer: customize Predefined Multiband Colors Create issue

### ISSUE THREAD

**Issue by aloboa** [EnMAP-Box/enmap-box#23]
Many of the current predefined buttons for color composites are quite redundant. It would be great advantage if these buttons could be easily customized using a user modifiable text file.

Meanwhile, is it possible for the user to modify the python script to redefine some of these buttons? which file should be modified?

**Comment by janzandr** [comment]
> Meanwhile, is it possible for the user to modify the python script to redefine some of these buttons? which file should be modified?

That is hard coded here:

![image](https://user-images.githubusercontent.com/45030715/188893836-98296a3c-3e21-4736-b0f7-ca9aa5e434fb.png)

**Comment by aloboa** [comment]
Thanks but that's for the raster Layer Styling (which is also pertinent).
I was talking about the Color Space Explorer and its Predefined Multiband Colors
(Natural color, False color, Color infrared etc)
![image](https://user-images.githubusercontent.com/6074054/188943958-1c10ec83-6ba7-49bd-8b13-839fe195c760.png)

which file would be the one to modify in this case?

**Comment by janzandr** [comment]
That is basically the same file. There is a hard mapping. If you change the center wavelength in the PY file, the RGB combinations will also be affected:
![image](https://user-images.githubusercontent.com/45030715/188950618-afaea910-c665-4c8d-9e77-e80e70e7c3b7.png)

**Comment by janzandr** [comment]
BTW - I plan to move all the functionality of the ColorSpaceExplorer into the RasterLayerStyling panel.

**Comment by aloboa** [comment]
But where is i.e the "vegetation analysis 1" defined? Perhaps I can change it there?
I favor the integration you mention, except if it results in a too complex panel.

**Comment by jakimowb** [comment]
I like the idea of use-based customization. To my experience, users / communities have a few well-defined bands / band combinations which they use frequently. From the large set of available bands shortcuts or band combinations, they often need only a small selection. For example, the water community is likely interested in different band selection shortcuts than the vegetation community.


I'd like to see the following features:
- allow to specify user-defined sets of pre-defined bands (shortcut name, target wavelength)
- allow to specify user-defined sets of RGB band combinations (shortcut name, 3 target wavelengths)

This could be done in a settings manager to specify all possible band / band-combination shortcuts and which of them should be visible in the raster layer styling panel.


This allows to reduce / filter the list of band shortcut buttons and combobox, which in this screenshot looks a bit crowded:
![image](https://user-images.githubusercontent.com/1404870/188977235-19af6b29-94c9-4270-aa54-c1e9711c4c5f.png)

to a smaller number, providing thematic/use-case specific simplicity.

**Comment by janzandr** [comment]
> I like the idea of use-based customization. To my experience, users / communities have a few well-defined bands / band combinations which they use frequently. From the large set of available bands shortcuts or band combinations, they often need only a small selection. For example, the water community is likely interested in different band selection shortcuts than the vegetation community.

> I'd like to see the following features:

> allow to specify user-defined sets of pre-defined bands (shortcut name, target wavelength)
allow to specify user-defined sets of RGB band combinations (shortcut name, 3 target wavelengths)
This could be done in a settings manager to specify all possible band / band-combination shortcuts and which of them should be visible in the raster layer styling panel.

> This allows to reduce / filter the list of band shortcut buttons and combobox, which in this screenshot looks a bit crowded:

Yes, that is the plan :-).

**Comment by janzandr** [comment]
> But where is i.e the "vegetation analysis 1" defined? Perhaps I can change it there?
I favor the integration you mention, except if it results in a too complex panel.

Sorry Agus, that is all a bit complicated and needs to be streamlined.
Currently, the RGBs are defined here in this *.ui file:
![image](https://user-images.githubusercontent.com/45030715/188985139-789e3bfb-b9ad-441c-8ad3-942485f505d4.png)

The RGB band names (e.g. B8-B11-B4) are Sentinel-2 bands, which are translated into the AwesomeSpectralIndices band names here:
![image](https://user-images.githubusercontent.com/45030715/188985743-92338330-4bd5-4552-ae0b-af248942364a.png)

Again, sorry, that needs to be simplified a lot!

**Comment by aloboa** [comment]
Just let you know that having these buttons to provide a range of specific color composites in a very simple way, was a great idea with potentially a great impact on people exploring remotely sensed images. But to be really useful,  these buttons should be customized.  I understand that it is not easy to design an efficient tool for this customization, but keep in mind that while not being sophisticated analysis, interactive exploring tools of imagery are most important. Actually, many people will do only that. Many thanks for your efforts in this direction.

**Comment by jakimowb** [comment]
From the implementation side it shouldn't be too complicated. 

The now hard-coded buttons in the *.ui file could be described in a JSON or XML file instead, which is loaded during widget initialization. The JSON may contain additional flags to specify which bands / band-combination should be shown in widgets like the raster layer styling panel or the color space explorer. This allows to show / hide buttons without deleting them form the JSON and add new ones.

A simple configuration tool to customize which buttons are to be shown could therefore be a simple text editor. However, a user-friendly configuration tool should offer a more sophisticated interface, like a table view to check/uncheck button visibility, change names, shortcut names, tooltips, etc.

**Comment by janzandr** [comment]
Yes, a JSON configuration file is a must have. A JSON for every standard sensor like Landsat, Sentinel-2, Modis etc. and as many user-def. JSON files as the user wants to.

A GUI may be a nice to have bonus.
